platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/TaskScheduler.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.common.wiring;
+
+import com.swirlds.common.wiring.counters.ObjectCounter;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.function.Consumer;
+
+/**
+ * Schedules tasks for a component.
+ *
+ * @param <OUT> the output type of the primary output wire (use {@link Void} if no output is needed)
+ */
+public abstract class TaskScheduler<OUT> extends OutputWire<OUT> {
```

Comment:
I don't think `TaskScheduler` should extend `OutputWire`. The proper relationship is that a Task Schedule has an output wire, not is an output wire.

Reply:
I will work on this change, not yet completed in the branch.

Reply:
change made